hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java

```diff
@@ -1647,16 +1661,8 @@ public synchronized boolean seekToNewSource(long targetPos)
     if (currentNode == null) {
       return seekToBlockSource(targetPos);
     }
-    boolean markedDead = dfsClient.isDeadNode(this, currentNode);
-    addToLocalDeadNodes(currentNode);
```

Comment:
Hi, @KeeProMise. The idea of this PR make sense to me. Leave some comments. Try to confirm whether we should delete those codes. I think we can not delete because this will destory dead node detector model. Should be like :

```java
  boolean markedDead = dfsClient.isDeadNode(this, currentNode);
  addToLocalDeadNodes(currentNode);
  if (!markedDead) {
      /* remove it from deadNodes. blockSeekTo could have cleared
       * deadNodes and added currentNode again. Thats ok. */
      removeFromLocalDeadNodes(oldNode);
    }
  DatanodeInfo newNode = blockSeekTo(targetPos, Collections.singletonList(currentNode));
  if (!currentNode.getDatanodeUuid().equals(newNode.getDatanodeUuid())) {
  xxx
  }
```
What your opinions?

Reply:
Thanks for your review, seekToNewSource is a synchronous method, so the method is atomic; and the method turns out to just add the current node to the local deadnodes; so I think it will not affect the dead node detector.

Reply:
Hi @KeeProMise , Let's think about below situation in original code:
we add currentNode to DFSInputStream#deadNodes,  this field has a getter method getLocalDeadNodes() which may be called by DFSClient and DFSInputStream's non-synchronized method.
But, your modification remove the logic of adding currentNode to DFSInputStream#deadNodes. The caller of getLocalDeadNodes() will be no aware of changing.

Reply:
hi @hfutatzhanghb, In all places where getLocal DeadNodes is used, currently only LocatedBlocksRefresher#refreshBlockLocations() is not synchronized. However, this method ( refreshBlockLocations() )determines whether the local deadnodes are empty. If not, the deadnodes will eventually be cleared. 
Therefore, not adding the current node to local dead nodes will not affect existing functions and is a safe operation. In addition, judging from the role of the seekToNewSource method, this method is to obtain a new data node to replace the current node. Therefore, the original method relies on local dead nodes, but is highly coupled with the dead nodes detector. This is also the reason why I proposed this PR.